dockerspawner/dockerspawner.py

```diff
@@ -1173,3 +1172,22 @@ def _fmt(v):
                 v = v["bind"]
             binds[_fmt(k)] = {"bind": _fmt(v), "mode": m}
         return binds
+
+    def _extra_create_kwargs_subs(self):
+        """
+        Substitutes values from template_namespace() for extra docker create kwargs.
+        """
+        subs = {}
+
+        def _fmt(v):
+            return v.format(**self.template_namespace())
+
+        for k, v in self.extra_create_kwargs.items():
+            if isinstance(v, dict):
+                subs[_fmt(k)] = {}
+                for k1, v1 in v.items():
+                    subs[_fmt(k)].update({_fmt(k1): _fmt(v1)})
```

Comment:
this will error if `v1` is not a string. It could itself be a list or dict or even a `docker.types` model.